test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/BusinessPartnerRecordServiceSupport.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers
+
+import cats.data.EitherT
+
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.onboarding.bpr.{BusinessPartnerRecordRequest, BusinessPartnerRecordResponse}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.services.onboarding.BusinessPartnerRecordService
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.Error
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait BusinessPartnerRecordServiceSupport {
+  this: ControllerSpec with SessionSupport =>
+
+  val mockBusinessPartnerRecordService = mock[BusinessPartnerRecordService]
+  def mockGetBusinessPartnerRecord(
+    businessPartnerRecordRequest: BusinessPartnerRecordRequest,
+    expectedBusinessPartnerRecordResponse: BusinessPartnerRecordResponse
+  ): Unit =
+    (mockBusinessPartnerRecordService
+      .getBusinessPartnerRecord(_: BusinessPartnerRecordRequest)(_: HeaderCarrier))
+      .expects(businessPartnerRecordRequest, *)
+      .returning(
+        EitherT[Future, Error, BusinessPartnerRecordResponse](
+          Future.successful(
+            Right(expectedBusinessPartnerRecordResponse)
```

Comment:
you can't be testing the error scenarios if you can't simulate one here

Reply:
Parameterized and added those missing tests too.